addons/osfstorage/models.py

```diff
@@ -255,6 +255,25 @@ def last_known_metadata(self):
             'last_seen': self.modified
         }
 
+    @property
+    def should_update_search(self):
+        target = self.target
+        # TODO: Can remove 'not file_.name' if we remove all base file nodes with name=None
+        file_node_is_qa = bool(
+            set(website_settings.DO_NOT_INDEX_LIST['tags']).intersection(self.tags.all().values_list('name', flat=True))
+        ) or bool(
+            set(website_settings.DO_NOT_INDEX_LIST['tags']).intersection(target.tags.all().values_list('name', flat=True))
+        ) or any(substring in target.title for substring in website_settings.DO_NOT_INDEX_LIST['titles'])
+        if not self.name or not target.is_public or file_node_is_qa or getattr(target, 'is_deleted', False) or \
+                getattr(target, 'archiving', False) or target.is_spam or self.is_deleted or getattr(target, 'is_retracted', False) or \
+                (target.spam_status == SpamStatus.FLAGGED and website_settings.SPAM_FLAGGED_REMOVE_FROM_SEARCH):
+            return False
```

Comment:
This is silly hard to read with the many many inline `or`s. Any good reason to not just have a bunch of cascading `ifs`? Will get the same short-circuit behavior.

I make no promises about the correctness of the following, but something like:
```
qa_tags = set(webiste_settings.DO_NOT_INDEX_LIST['tags']
if qa_tags.intersection(self.tags.all().values_list('name', flat=True):
  return False
if qa_tags.intersection(target.tags.all().values_list('name', flat=True):
  return False
if any(qa_substring in target.title for qa_substring in website_settings.DO_NOT_INDEX_LIST['titles']):
  return False
if not self.name:
  return False

spam_check_field = (
  'is_spammy'
  if website_settings.SPAM_FLAGGED_REMOVE_FROM_SEARCH 
  else 'is_spam'
)
target_check_fields = [
  ('is_public', True),
  ('deleted', Null),
  ('is_deleted' False),
  ('is_retracted', False),
  (spam_check_field, False),
  ('archiving', False),
  ('verified_publishable', True),
  ('primary_file', self)
]
for field_name, expected_value in target_check_fields:
  if getattr(target, field_name, expected_value) != expected_value:
    return False
```

Reply:
(note, I see where this was copied from now)